mantidproject/mantid
# Remove unnecessary null pointer checks

**Description of work.**

Removed unnecessary checks for nullptr before calling delete

**To test:**

Ensure all unit tests pass

Fixes #23103 

Does this update require release notes?


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
